images/access_setup/content/bin/setup_kcp.sh

```diff
@@ -0,0 +1,156 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR="$(
+  cd "$(dirname "$0")" >/dev/null
+  pwd
+)"
+
+source "$SCRIPT_DIR/common.sh"
+
+usage() {
+  printf "Usage:
+    %s [options]
+
+Generate access credentials for a new KCP instance so it can be managed by pipelines as code
+
+Optional arguments:
+    -k, --kubeconfig KUBECONFIG
+        kubeconfig to the kcp instance to configure.
+        The current context will be used.
+        Default value: \$KUBECONFIG
+    --kcp_org KCP_ORG
+        Path to the organization workspace.
+        Can be read from \$KCP_ORG.
+        Default value: Workspace in use by the current context.
+    --kcp_workspace KCP_WORKSPACE
+        Name of the workspace in which the compute clusters will be registered.
+        Can be read from \$KCP_WORKSPACE.
+        Default: compute
+    -w, --work_dir WORK_DIR
+        Directory into which the credentials folder will be created.
+        Default: ./work
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    %s -d -k /path/to/kcp.kubeconfig
+" "${0##*/}" "${0##*/}" >&2
+}
+
+parse_args() {
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+    -k | --kubeconfig)
+      shift
+      KUBECONFIG="$1"
+      ;;
+    --kcp_org)
+      shift
+      KCP_ORG="$1"
+      ;;
+    --kcp_workspace)
+      shift
+      KCP_WORKSPACE="$1"
+      ;;
+    -w | --work_dir)
+      shift
+      WORK_DIR="$1"
+      ;;
+    -d | --debug)
+      set -x
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    --)
+      # End of arguments
+      break
+      ;;
+    *)
+      exit_error "Unknown argument: $1\n"
+      ;;
+    esac
+    shift
+  done
+}
+
+prechecks() {
+  KUBECONFIG=${KUBECONFIG:-}
+  if [[ -z "${KUBECONFIG}" ]]; then
+    exit_error "Missing parameter --kubeconfig"
+  fi
+  if [[ ! -f "$KUBECONFIG" ]]; then
+    echo "File not found: $KUBECONFIG" >&2
+    exit 1
+  fi
+  export KUBECONFIG
+
+  WORK_DIR=${WORK_DIR:-./work}
+
+  KCP_ORG=${KCP_ORG:-$(kubectl kcp workspace current | cut -d\" -f2 | cut -d: -f1,2)}
+  KCP_WORKSPACE=${KCP_WORKSPACE:-compute}
+}
+
+init() {
+  WORK_DIR="${WORK_DIR:-$SCRIPT_DIR/work}"
+  mkdir -p "$WORK_DIR"
+
+  credentials_dir="$WORK_DIR/credentials/kubeconfig"
+  mkdir -p "$credentials_dir"
+}
+
+generate_kcp_credentials() {
+  kcp_name="$(kubectl config view --minify -o json |
+    jq ".clusters[0].cluster.server" |
+    sed -r -e "s@.*://([^.]*).*@\1@").$(
+    kubectl kcp workspace current | cut -d\" -f2 | cut -d: -f2- | tr ':' '.'
+  )"
+  printf "[KCP: %s\n]" "$kcp_name"
```

Comment:
Should be : printf "[KCP: %s]\n" "$kcp_name"

Reply:
Fixed